ui/app/components/tools/hash.ts

```diff
@@ -17,32 +21,33 @@ import errorMessage from 'vault/utils/error-message';
  * <Tools::Hash />
  */
 export default class ToolsHash extends Component {
-  @service store;
-  @service flashMessages;
+  @service declare readonly api: ApiService;
+  @service declare readonly flashMessages: FlashMessageService;
 
   @tracked algorithm = 'sha2-256';
   @tracked format = 'base64';
```

Comment:
For my own information, I know these files are older, but wouldn't these defaults be set by the model? And now that we're moving away from a model, they'd be set by the apiSpec? For example [algorithm's](https://developer.hashicorp.com/vault/api-docs/system/tools#hash-data) default would be set by the api as 'sha2-256'

Reply:
In this case there are no models for tools so it was just using the adapter and directly making a request. You bring up a good question about defaults. I can see that the default values for this request are defined in the OpenAPI spec, but I'm not seeing anywhere in the client where it is applying the values. I suppose that it doesn't bother populating the values if they are not required properties since the server will default the values on that end.